daemon/emer-machine-id-provider.c

```diff
@@ -291,24 +296,32 @@ emer_machine_id_provider_get_id (EmerMachineIdProvider *self,
   EmerMachineIdProviderPrivate *priv =
     emer_machine_id_provider_get_instance_private (self);
 
-  static gboolean id_is_valid = FALSE;
-  G_LOCK_DEFINE_STATIC (id_is_valid);
-
-  G_LOCK (id_is_valid);
-  if (!id_is_valid)
+  /* The assumption is that this will not be called
+   * multiple times for the same instance across different
+   * threads. */
+  if (!priv->id_is_valid)
     {
       if (read_machine_id (self))
         {
-          id_is_valid = TRUE;
+          priv->id_is_valid = TRUE;
         }
       else
         {
-          G_UNLOCK (id_is_valid);
           return FALSE;
         }
     }
-  G_UNLOCK (id_is_valid);
 
   uuid_copy (machine_id, priv->id);
   return TRUE;
 }
+
+void
+emer_machine_id_provider_reload (EmerMachineIdProvider *self)
+{
+  EmerMachineIdProviderPrivate *priv =
+    emer_machine_id_provider_get_instance_private (self);
+
+  g_message ("EmerMachineIdProvider: Reloading from %s", priv->path);
```

Comment:
This message isn't true – that file will not be read until an unspecified time in the future.

Reply:
That's true. I suppose I could rephrase it to "will reload from %s"